charts/kube-prometheus-stack/README.md

```diff
@@ -83,6 +83,17 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### From 21.x to 22.x
+
+Due to the upgrade of the `kube-state-metrics` chart, removal of its deployment/stateful needs to done manually prior to upgrading:
+```console
+kubectl delete deployments.apps -l app.kubernetes.io/instance=prometheus-operator,app.kubernetes.io/name=kube-state-metrics --cascade=orphan
+```
+or
+```console
+kubectl delete statefulsets.apps -l app.kubernetes.io/instance=prometheus-operator,app.kubernetes.io/name=kube-state-metrics --cascade=orphan
```

Comment:
I can't find an option in kube-state-metrics chart to run it as statefulset?

https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-state-metrics

So why do we need that other option?

Reply:
See: https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-state-metrics/templates/deployment.yaml#L1-L6

It depends on the `autosharding` setting.

Reply:
Sorry, web frontend commit messed up cto check. Will fix tomorrow or feel free to do it by yourself.